inc/namespace.php

```diff
@@ -954,3 +960,43 @@ function set_wp_debug_constants() : void {
 	}
 	define( 'WP_DEBUG_DISPLAY', false );
 }
+
+/**
+ * Log a message to our cloud services. This will use Fluent Bit if it is
+ * available, otherwise log directly to CloudWatch..
+ *
+ * @param string $log_group Name of the log group to send logs to. Environment
+ *                          name is added automatically.
+ * @param string $log_stream Name of the log stream.
+ * @param string $message Message to log
+ * @param string $level Level of the log. Corresponds to Psr\Log\LoggerInterface
+ * @see https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md#3-psrlogloggerinterface
+ * @return boolean
+ */
+function log_to_cloud( string $log_group, string $log_stream, string $message, string $level = 'info' ) : bool {
+	if ( ! Fluent_Bit\available() ) {
+		$result = CloudWatch_Logs\send_events_to_stream(
+			[
+				[
+					'timestamp' => time() * 1000,
+					'message' => $message,
+				],
+			],
+			Altis\get_environment_name() . '/' . $log_group,
+			$log_stream
+		);
+
+		return $result;
+	}
+
+	// Basic error correction to ensure $level is an allowed value.
+	if ( ! in_array( $level, [ 'debug', 'info', 'notice', 'warning', 'error', 'critical', 'alert', 'emergency' ], true ) ) {
+		$level = 'info';
+	}
+
+	$fluent_tag = sprintf( 'app.%s.%s', $log_group, $log_stream );
+	$logger = Fluent_Bit\get_logger( $fluent_tag );
+	$logger->$level( $message );
```

Comment:
Is there a generic `log` function that takes the level as a param? thinking that might be cleaner than a dynamically called method.

Reply:
Bah, there totally is 🤦 Good call 👍